core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java

```diff
@@ -265,18 +265,12 @@ public String toString() {
   }
 
   @Override
-  public String getFingerprint(String path) throws IOException {
-    return call(new UfsCallable<String>() {
-      @Override
-      public String call() throws IOException {
-        return mUnderFileSystem.getFingerprint(path);
-      }
-
-      @Override
-      public String toString() {
-        return String.format("GetFingerprint: path=%s", path);
-      }
-    });
+  public String getFingerprint(String path) {
+    String method = String.format("GetFingerprint: path=%s", path);
+    LOG.debug("Enter: {}", method);
+    String ret = mUnderFileSystem.getFingerprint(path);
+    LOG.debug("Exit (OK): {}", method);
+    return ret;
```

Comment:
Could you still use the helper method? This isolates the logging to a single place so we can update it. If the exception signature is the problem, you could do a try catch?

Reply:
Done